src/mame/nintendo/n64_v.cpp

```diff
@@ -1358,368 +1360,370 @@ void n64_rdp::disassemble(uint64_t *cmd_buf, char* buffer)
-		default:    sprintf(buffer, "Unknown (%08X %08X)", uint32_t(cmd_buf[0] >> 32), (uint32_t)cmd_buf[0]); break;
+		case 0x26:  buffer << "Sync_Load"; break;
+		case 0x27:  buffer << "Sync_Pipe"; break;
+		case 0x28:  buffer << "Sync_Tile"; break;
+		case 0x29:  buffer << "Sync_Full"; break;
+		case 0x2d:  util::stream_format(buffer, "Set_Scissor            %s, %s, %s, %s", sl, tl, sh, th); break;
+		case 0x2e:  util::stream_format(buffer, "Set_Prim_Depth         %04X, %04X", uint32_t(cmd_buf[0] >> 16) & 0xffff, (uint32_t)cmd_buf[0] & 0xffff); break;
+		case 0x2f:  util::stream_format(buffer, "Set_Other_Modes        %08X %08X", uint32_t(cmd_buf[0] >> 32), (uint32_t)cmd_buf[0]); break;
+		case 0x30:  util::stream_format(buffer, "Load_TLUT              %d, %s, %s, %s, %s", tile, sl, tl, sh, th); break;
+		case 0x32:  util::stream_format(buffer, "Set_Tile_Size          %d, %s, %s, %s, %s", tile, sl, tl, sh, th); break;
+		case 0x33:  util::stream_format(buffer, "Load_Block             %d, %03X, %03X, %03X, %03X", tile, uint32_t(cmd_buf[0] >> 44) & 0xfff, uint32_t(cmd_buf[0] >> 32) & 0xfff, uint32_t(cmd_buf[0] >> 12) & 0xfff, uint32_t(cmd_buf[0]) & 0xfff); break;
+		case 0x34:  util::stream_format(buffer, "Load_Tile              %d, %s, %s, %s, %s", tile, sl, tl, sh, th); break;
+		case 0x35:  util::stream_format(buffer, "Set_Tile               %d, %s, %s, %d, %04X", tile, format, size, (uint32_t(cmd_buf[0] >> 41) & 0x1ff) * 8, (uint32_t(cmd_buf[0] >> 32) & 0x1ff) * 8); break;
+		case 0x36:  util::stream_format(buffer, "Fill_Rectangle         %s, %s, %s, %s", sh, th, sl, tl); break;
+		case 0x37:  util::stream_format(buffer, "Set_Fill_Color         R: %d, G: %d, B: %d, A: %d", r, g, b, a); break;
+		case 0x38:  util::stream_format(buffer, "Set_Fog_Color          R: %d, G: %d, B: %d, A: %d", r, g, b, a); break;
+		case 0x39:  util::stream_format(buffer, "Set_Blend_Color        R: %d, G: %d, B: %d, A: %d", r, g, b, a); break;
+		case 0x3a:  util::stream_format(buffer, "Set_Prim_Color         %d, %d, R: %d, G: %d, B: %d, A: %d", uint32_t(cmd_buf[0] >> 40) & 0x1f, uint32_t(cmd_buf[0] >> 32) & 0xff, r, g, b, a); break;
+		case 0x3b:  util::stream_format(buffer, "Set_Env_Color          R: %d, G: %d, B: %d, A: %d", r, g, b, a); break;
+		case 0x3c:  util::stream_format(buffer, "Set_Combine            %08X %08X", uint32_t(cmd_buf[0] >> 32), (uint32_t)cmd_buf[0]); break;
+		case 0x3d:  util::stream_format(buffer, "Set_Texture_Image      %s, %s, %d, %08X", format, size, (uint32_t(cmd_buf[0] >> 32) & 0x1ff) + 1, (uint32_t)cmd_buf[0]); break;
+		case 0x3e:  util::stream_format(buffer, "Set_Mask_Image         %08X", (uint32_t)cmd_buf[0]); break;
+		case 0x3f:  util::stream_format(buffer, "Set_Color_Image        %s, %s, %d, %08X", format, size, (uint32_t(cmd_buf[0] >> 32) & 0x1ff) + 1, (uint32_t)cmd_buf[0]); break;
+		default:    util::stream_format(buffer, "Unknown (%08X %08X)", uint32_t(cmd_buf[0] >> 32), (uint32_t)cmd_buf[0]); break;
 	}
+
+	return buffer.str();
```

Comment:
You should use `return std::move(buffer).str();` so when we move to C++20 it can use the destructive variant of `.str()` since it won’t use `buffer` again.

Reply:
Interesting. I've never seen this before.
Done

Reply:
> Interesting. I've never seen this before. Done

If you look at the overloads of that function here: https://en.cppreference.com/w/cpp/io/basic_ostringstream/str

C++20 introduces an overload for extracting the contents destructively (number 3, the one with the `&&` at the end).  Various things will prevent us from moving to C++20 for a while, but it’s harmless to do little things like this to prepare for it when the time comes.